crates/anvil/core/src/eth/transaction/optimism.rs

```diff
@@ -145,7 +328,7 @@ impl DepositTransaction {
     /// - `input`
     pub fn decode_inner(buf: &mut &[u8]) -> Result<Self, DecodeError> {
         Ok(Self {
-            nonce: Decodable::decode(buf)?,
+            nonce: U256::ZERO,
```

Comment:
was this a bug?
I don't understand why this is not decoded

Reply:
Yeah, deposit transactions do not have a nonce and always default to 0. See [spec](https://github.com/ethereum-optimism/specs/blob/main/specs/deposits.md#the-deposited-transaction-type)